app/controllers/providers_controller.rb

```diff
@@ -95,6 +96,21 @@ def search
     redirect_to provider_path(provider_code)
   end
 
+  def training_providers_courses
```

Comment:
I think it would be simpler, and more in keeping with the conventions, for this to an index action on another controller e.g. `TrainingProviderCoursesController`

Reply:
done